shell/components/ResourceDetail/Masthead.vue

```diff
@@ -384,7 +384,7 @@ export default {
             </nuxt-link>
             <span v-else>{{ parent.displayName }}:</span>
             <span v-if="value.detailPageHeaderActionOverride && value.detailPageHeaderActionOverride(realMode)">{{ value.detailPageHeaderActionOverride(realMode) }}</span>
-            <t v-else :k="'resourceDetail.header.' + realMode" :subtype="resourceSubtype" :name="displayName" />
+            <t v-else :k="'resourceDetail.header.' + realMode" :subtype="resourceSubtype" :name="displayName" :escapehtml="false" />
```

Comment:
I don't think you have to add a prop to the i18n mixin to render the name in a way that matches the list view, as required in the ticket https://github.com/harvester/harvester/issues/2563. This should render the same way:
```
<span v-else>
   {{ t(`resourceDetail.header.${realMode}`, {subtype: resourceSubtype, name:displayName}, {raw:true}) }}
</span>
```

Reply:
@mantis-toboggan-md  This is very strange.
1. The `raw` parameter indicates that we want it to convert the html correctly.   (But here we don't want it to convert, the html stays as it is)，   
2. `{raw: true} `should be changed to `true`， https://github.com/rancher/dashboard/blob/a71dee857effc74995b4638fa816f8af1138c17e/shell/plugins/i18n.js#L27

Reply:
The question here is, how should I interpret the `raw` parameter inside [Vue.prototype.t](https://github.com/rancher/dashboard/blob/a71dee857effc74995b4638fa816f8af1138c17e/shell/plugins/i18n.js#L27). 
If `raw` here is the same as `raw` in [Vue.component](https://github.com/rancher/dashboard/blob/a71dee857effc74995b4638fa816f8af1138c17e/shell/plugins/i18n.js#L89), then I think it makes sense to add `escapehtml` prop,

Reply:
> {raw: true} should be changed to true，

You're absolutely right, I apologize for being confusing. The way I wrote it is wrong but "worked" because of how this if statement is written:
https://github.com/rancher/dashboard/blob/a71dee857effc74995b4638fa816f8af1138c17e/shell/plugins/i18n.js#L20-L24

My solution works because [Vue interprets expressions in `{{}}` as plain text](https://v2.vuejs.org/v2/guide/syntax.html#Raw-HTML). So these two examples do not render the same way:
```
<t v-else :k="'resourceDetail.header.' + realMode" :subtype="resourceSubtype" :name="displayName" raw />
<span v-else>
  {{ t(`resourceDetail.header.${realMode}`, {subtype:resourceSubtype, name: displayName}, true) }}
</span>
  ```
` <t raw>` will not call `escapeHtml` on the translation string, and will set the element's innerHTML to include raw (aka unescaped) html. That would render the example text `<strong><em>something_interesting</em></strong>` as **_something_interesting_** https://github.com/rancher/dashboard/blob/a71dee857effc74995b4638fa816f8af1138c17e/shell/plugins/i18n.js#L89-L94
When we use `t(<translation key>, <translation args>, true)` inside `{{}}`, `t` returns a string with HTML unescaped, but the html is rendered as plain text by vue anyway. I think this is also why, in the original issue, we saw html tags as plain text in the list view:
https://github.com/rancher/dashboard/blob/a71dee857effc74995b4638fa816f8af1138c17e/shell/components/formatter/LinkDetail.vue#L45-L47

`value` in the issue example is `"<strong><em>something_interesting</em></strong>"` but then, because it is in `{{}}`, it is rendered as plain text.

I think needing to render a translation containing user input is rare, so I didn't want to add a prop to i18n mixin to handle this one situation. Having said that I understand that my solution relies on a detail of Vue implementation instead of something we have explicitly written in the codebase, so if it is not clear enough and you want to keep the prop I'm okay with that.

Reply:
Yes, I think it might be good to keep this prop.